### DISCUSSION TITLE
Additional meta data with a blog post

### DISCUSSION THREAD

**Discussion by pitosalas** [squidfunk/mkdocs-material#6733] [General]
I am using material and the blog plugin. As you can see from the screenshot, I get nice meta data with the post. I get the posting date and the minutes read. What if I want to add more meta data? For example one or more urls?

<img width="1025" alt="Screenshot 2024-02-02 at 1 16 27 PM" src="https://github.com/squidfunk/mkdocs-material/assets/44681/52919777-659e-401f-8cea-0701c70d03cc">

**Comment by kamilkrzyskow** [comment]
Hello @pitosalas,
in the community version of the Blog plugin only the creation date, updated date, categories and read time are available as the metadata. 
In the Insiders version of the plugin it's mostly the same, there is also the option to add [related reference links](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#adding-related-links)

If you need more you'd have to use theme extension: https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
You'd have to override the `blog-post.html` template and add more metadata extraction as you see fit:
https://github.com/squidfunk/mkdocs-material/blob/02bf848fd504b953acb1bdb8aad9531ac052c93a/src/templates/blog-post.html#L77-L153

**Reply by squidfunk** [reply]
Related: https://github.com/squidfunk/mkdocs-material/discussions/4397#discussioncomment-8058936

**Reply by pitosalas** [reply]
Looks like I have to copy the whole of blog-post.html to overrides because it doesn't have "blocks", am I understanding that right?

**Reply by pitosalas** [reply]
And in that code, how do I retrieve for example a "markdown front matter variable called: "original"?

**Reply by kamilkrzyskow** [reply]
> Looks like I have to copy the whole of blog-post.html to overrides because it doesn't have "blocks", am I understanding that right?

Yes, the `blog-post.html` extends `main.html` and everything is already an override of the `container` block.

> And in that code, how do I retrieve for example a "markdown front matter variable called: "original"?

In normal templates you'd do something like `page.meta.original`, but based on the example above you should access it via `page.config.original`. Remember to use the special brackets `{{ }}`, when accessing Jinja variables.

EDIT: 
After looking at the code, I can say that `page.meta.original` would still work, as the `meta` property isn't cleared, but is merged into `config` for convenience (?) 
https://github.com/squidfunk/mkdocs-material/blob/bcad9cecd18b47409988ebb0aa69bea6a7edd70c/src/plugins/blog/structure/__init__.py#L90-L98

**Reply by squidfunk** [reply]
> Looks like I have to copy the whole of blog-post.html to overrides because it doesn't have "blocks", am I understanding that right?

Correct, because the `blog-post.html` is a partial and we cannot use blocks in partials. Furthermore, we're planning to cut it up into smaller partials and pieces, but that's left to the next iteration on the blog plugin which currently does not have an ETA but will definitely come. Partials might change significantly, because the blog plugin and templates are currently still experimental.

**Reply by squidfunk** [reply]
> After looking at the code, I can say that page.meta.original would still work, as the meta property isn't cleared, but is merged into config for convenience (?)

We use MkDocs configuration validation features to validate the blog post meta data, so `post.config` contains the validated and potentially transformed metadata. You can still use `post.meta` in templates, which is the raw input.

**Comment by pitosalas** [comment]
Thanks all. I did get it to work just the way I want!

Pito
